esrally/track/loader.py

```diff
@@ -757,9 +757,10 @@ class TrackSpecificationReader:
     Creates a track instances based on its parsed JSON description.
     """
 
-    def __init__(self, override_auto_manage_indices=None, source=io.FileSource):
+    def __init__(self, override_auto_manage_indices=None, track_params=None, source=io.FileSource):
         self.name = None
         self.override_auto_manage_indices = override_auto_manage_indices
+        self.track_params = track_params if track_params else {}
```

Comment:
Completely minor question, but could this be replaced by `self.track_params = track_params` if the above named parameter had the default value set to `track_params = {}`?

Reply:
No, this is not possible because the default argument is mutable. This is a [common gotcha](http://docs.python-guide.org/en/latest/writing/gotchas/#mutable-default-arguments).